locator/types.hh

```diff
@@ -21,12 +25,101 @@ namespace locator {
 
 using inet_address = gms::inet_address;
 
+class token_metadata;
+class topology;
+
+struct host_id_or_endpoint {
+    host_id id;
+    gms::inet_address endpoint;
+
+    enum class param_type {
+        host_id,
+        endpoint,
+        auto_detect
+    };
```

Comment:
std::variant?

It smells dubious, but maybe I'm missing something.

Reply:
We're passing a string to parse.  `param_type` tells the constructor what to expect so it can reject invalid input (sent from the api handler)